### PR TITLE
Implemented endpoints for get trust and get estate

### PR DESCRIPTION
**Pull request by joepowell96**
Implemented 2 endpoints, one for getting a trust and one for getting an estate.
Added explicit auditing for the responses for the two endpoints.

### CODE REVIEW

File: app/uk/gov/hmrc/trusts/models/get_trust_or_estate/get_estate/GetEstateSchemaModels.scala [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  )(GetEstate.apply _)
+}
+
+case class GetEstateDesResponse(getEstate: Option[GetEstate],
+                               responseHeader: ResponseHeader)
+
+object GetEstateDesResponse {
+  implicit val writes: Writes[GetEstateDesResponse] = Json.writes[GetEstateDesResponse]
+  implicit val reads: Reads[GetEstateDesResponse] = (
+    (JsPath \ "estateOrEstateDisplay").readNullable[GetEstate] and
```
Comment (jassalrichy) [inline app/uk/gov/hmrc/trusts/models/get_trust_or_estate/get_estate/GetEstateSchemaModels.scala]: estateOrEstateDisplay ? ... should this not be trustOrEstateDisplay ?

Comment (adamconder) [reply app/uk/gov/hmrc/trusts/models/get_trust_or_estate/get_estate/GetEstateSchemaModels.scala]: 👍 I think it should be as well. Let's check the spec.

File: app/uk/gov/hmrc/trusts/controllers/GetEstateController.scala [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.trusts.controllers
+
+import javax.inject.{Inject, Singleton}
+import play.api.libs.json.Json
+import play.api.mvc.{Action, AnyContent}
+import uk.gov.hmrc.play.bootstrap.controller.BaseController
+import uk.gov.hmrc.trusts.controllers.actions.{IdentifierAction, ValidateUTRAction}
```
Comment (adamconder) [inline app/uk/gov/hmrc/trusts/controllers/GetEstateController.scala]: Maybe something to refactor later, I'd like to keep Identifier and Validate the same in the sense they both have an `apply()` function and use DI to inject them into the controller.

File: app/uk/gov/hmrc/trusts/models/get_trust_or_estate/ResponseModels.scala [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.trusts.models.get_trust_or_estate
+
+import uk.gov.hmrc.trusts.models.get_trust_or_estate.get_estate.GetEstateResponse
+import uk.gov.hmrc.trusts.models.get_trust_or_estate.get_trust.GetTrustResponse
+
+trait ErrorResponse
```
Comment (adamconder) [inline app/uk/gov/hmrc/trusts/models/get_trust_or_estate/ResponseModels.scala]: Can this be a sealed trait?

File: test/uk/gov/hmrc/trusts/services/DesServiceSpec.scala [hunk 4]
```diff
@@ -279,5 +283,205 @@ class DesServiceSpec extends BaseSpec {
     }
   } //getSubscriptionId
 
+  ".getTrustInfo" should {
```
Comment (adamconder) [inline test/uk/gov/hmrc/trusts/services/DesServiceSpec.scala]: Do we have tests where the response is `Processed` with the `Trust/Estate` and all entities returned in the response?

File: app/uk/gov/hmrc/trusts/models/get_trust_or_estate/get_estate/GetEstateSchemaModels.scala [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+import play.api.libs.functional.syntax._
+import play.api.libs.json._
+import uk.gov.hmrc.trusts.models.get_trust_or_estate.{MatchData, ResponseHeader}
+import uk.gov.hmrc.trusts.models.{Correspondence, Declaration, Estate}
+
+case class GetEstate(matchData: MatchData,
+                    correspondence: Correspondence,
+                    declaration: Declaration,
+                    estate: Estate)
```
Comment (adamconder) [inline app/uk/gov/hmrc/trusts/models/get_trust_or_estate/get_estate/GetEstateSchemaModels.scala]: This won't work for `Trust/Estate` as the entity types within this are not mapping `lineNo`, `bpMatchStatus` and `startDate` for each entity.

**Pull request merged**